### DISCUSSION TITLE
How to test a type's optional param

### DISCUSSION THREAD

**Discussion by IDrumsey** [vitest-dev/vitest#5203] [Q&A]
I have the following type

```typescript
export type APIResponseError = {
    code?: number,
    message: string
}
```

I'm trying to test with

```typescript
// valid assignment
expectTypeOf({code: 123, message: "asdf"}).toEqualTypeOf<APIResponseError>() // showing error when I don't want it to
```

I know it throws the error because code isn't undefinable in the actual type, I'm just not sure how to fix this.

**Comment by sheremet-va** [comment]
code is string in your type, but number in your expect statement - it fails because of that

**Reply by IDrumsey** [reply]
whoops, my bad, it was supposed to be number. I updated the question.

**Comment by hi-ogawa** [comment]
Did you also check upstream's doc? https://github.com/mmkal/expect-type
It looks like you want an assignability than equivalence, so `toMatchTypeOf` is what you looking for? https://vitest.dev/api/expect-typeof.html#tomatchtypeof
(Or you could also use `{ code: 123, message: "asdf" } satisfies APIResponseError`)
